Packs/AzureKeyVault/Integrations/AzureKeyVault/AzureKeyVault_test.py

```diff
@@ -0,0 +1,452 @@
+
+    command_arguments = {'vault_name': VAULT_NAME}
+    result = list_secrets_command(mock_client(), command_arguments)
+    assert len(result.outputs) == 1
+    assert result.outputs_prefix == SECRET_PREFIX
+    assert result.outputs[0].get('id') == "https://myvault.vault.azure.net/secrets/listsecrettest0"
+    assert result.outputs[0].get('contentType') == "plainText"
+    assert result.outputs[0].get('attributes').get('enabled') is True
+
+
+def test_azure_key_vault_secret_get_command(requests_mock):
+    """
+    Scenario: get secret.
+    Given:
+     - User has provided valid credentials.
+     - Key vault name
+     - Secret name
+    When:
+     - azure-key-vault-secret-get command called.
+    Then:
+     - Ensure number of items is correct.
+     - Ensure outputs prefix is correct.
+     - Ensure a sample value from the API matches what is generated in the context.
+    """
+
+    mock_response = json.loads(load_mock_response('get_secret.json'))
+    url = f'{BASE_VAULT_URL}/secrets/{SECRET_NAME}{API_VAULT_VERSION_PARAM}'
+
+    requests_mock.post(ACCESS_TOKEN_REQUEST_URL, json=mock_response)
+    requests_mock.get(url, json=mock_response)
+
+    command_arguments = {'vault_name': VAULT_NAME,
+                         'secret_name': SECRET_NAME}
+    result = get_secret_command(mock_client(), command_arguments)
+    assert len(result.outputs) == 7
+    assert result.outputs_prefix == SECRET_PREFIX
+    assert result.outputs.get('value') == "mysecretvalue"
+    assert result.outputs.get('kid') == "mykid"
+    assert result.outputs.get('key_vault_name') == VAULT_NAME
+    assert result.outputs.get('attributes').get('enabled') is True
+
+
+def test_azure_key_vault_secret_delete_command(requests_mock):
+    """
+    Scenario: delete secret.
+    Given:
+     - User has provided valid credentials.
+     - Key vault name
+     - Secret name
+    When:
+     - azure-key-vault-secret-delete command called.
+    Then:
+     - Ensure number of items is correct.
+     - Ensure outputs prefix is correct.
+     - Ensure a sample value from the API matches what is generated in the context.
+    """
+
+    mock_response = json.loads(load_mock_response('delete_secret.json'))
+    url = f'{BASE_VAULT_URL}/secrets/{SECRET_NAME}{API_VAULT_VERSION_PARAM}'
+
+    requests_mock.post(ACCESS_TOKEN_REQUEST_URL, json=mock_response)
+    requests_mock.delete(url, json=mock_response)
+
+    command_arguments = {'vault_name': VAULT_NAME,
+                         'secret_name': SECRET_NAME}
+    result = delete_secret_command(mock_client(), command_arguments)
+    assert len(result.outputs) == 6
+    assert result.outputs_prefix == 'AzureKeyVault.Secret'
+    assert result.outputs.get('recoveryId') == "https://test.vault.azure.net/deletedsecrets/sec_test"
+    assert result.outputs.get('key_vault_name') == VAULT_NAME
+    assert result.outputs.get('deletedDate') == "2017-05-04T22:53:53"
+
+
+def test_azure_key_vault_certificate_list_command(requests_mock):
```

Comment:
let's add tests not just for commands, but also (and mainly) for helper functions and their pythonic logic. This is one of the main goals of unit tests.